src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4822,6 +4573,42 @@ private static void AddUniqueVariable(HashSet<string> hashedResults, List<Comple
             }
         }
 
+        private static string GenerateVariableCompletionText(string value)
+        {
+            return GenerateVariableCompletionText(value, true);
+        }
+
+        private static string GenerateVariableCompletionText(string value, bool hasProvider)
+        {
+            if (string.IsNullOrEmpty(value))
+                return string.Empty;
+
+            bool startsWithQM = value[0] == '?' && value.Length > 1;
+            bool lastCharWasColon = false, hasColon = false;
+            bool braceVariable = !value[0].IsVariableStart() || value.Length != 1 && (value[0] == '$' || value[0] == '^' || (hasProvider && startsWithQM));
+            if (!braceVariable)
+            {
+                lastCharWasColon = !hasProvider && value[0] == ':';
```

Comment:
I just realized a major flaw in my thought process regarding `hasColon` / `lastCharWasColon`, as I was trying to perform two tests, a check for double colons, and regardless of braced or not, if there was any colons in the entire name.  This loop is not set up for that.   Since I need to scan the entire name for a colon, it appears its most efficient to use `Contains(char)`, than it is to make the current loop continue scanning for the colon even if it has found the brace condition.  Unfortunately it still needs to look for `::` as one of the possible brace conditions,